nautobot/utilities/templatetags/buttons.py

```diff
@@ -42,14 +42,11 @@ def clone_button(instance):
 
 
 @register.inclusion_tag("buttons/edit.html")
-def edit_button(instance, use_pk=False):
+def edit_button(instance, key="slug"):
```

Comment:
Thinking about this a bit more, it might be good from a backwards-compatibility standpoint (e.g. for plugins) to still have a `use_pk` option - is that feasible to implement? Something along the lines of:

```python
def edit_button(instance, use_pk=False, key="slug"):
   """TODO add docstring here explaining that use_pk is deprecated, etc."""
   viewname = _get_viewname(instance, "edit")
   
   # Assign kwargs
   if use_pk:
       kwargs = {"pk": instance.pk}
   else:
       kwargs = {key: getattr(instance, key)}
```